sql/src/main/java/io/crate/expression/reference/information/TablesSettingsExpression.java

```diff
@@ -155,10 +159,47 @@ private void addChildImplementations() {
 
         static final String ENABLE = "enable";
         static final String TOTAL_SHARDS_PER_NODE = "total_shards_per_node";
+        static final String REQUIRE = "require";
+        static final String INCLUDE = "include";
+        static final String EXCLUDE = "exclude";
 
         private void addChildImplementations() {
             childImplementations.put(ENABLE, new StringTableParameterExpression(TableParameterInfo.ROUTING_ALLOCATION_ENABLE.getKey()));
             childImplementations.put(TOTAL_SHARDS_PER_NODE, new TableParameterExpression(TableParameterInfo.TOTAL_SHARDS_PER_NODE.getKey()));
+            childImplementations.put(REQUIRE, new DynamicObjectCollectionExpression(IndexMetaData.INDEX_ROUTING_REQUIRE_GROUP_PREFIX));
+            childImplementations.put(INCLUDE, new DynamicObjectCollectionExpression(IndexMetaData.INDEX_ROUTING_INCLUDE_GROUP_PREFIX));
+            childImplementations.put(EXCLUDE, new DynamicObjectCollectionExpression(IndexMetaData.INDEX_ROUTING_EXCLUDE_GROUP_PREFIX));
+        }
+    }
+
+    static class DynamicObjectCollectionExpression extends NestableCollectExpression<RelationInfo, Map<String, Object>> {
+
+        private final String name;
+        private Map<String, Object> value;
+
+        DynamicObjectCollectionExpression(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public void setNextRow(RelationInfo row) {
+            if (row instanceof DocTableInfo) {
```

Comment:
would it make sense to extract this into a static pure function and use `NestableCollectExpression.forFunction` ?